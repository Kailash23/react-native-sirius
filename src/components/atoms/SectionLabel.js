import React from 'react';
import { Text, withTheme } from 'react-native-paper';
import { Platform } from 'react-native';
import { customFonts } from '@styles/customFonts';
import { SIZES, HalfCardImageWidth, getFontSize } from '@styles/sizes';

function SectionLabel(props) {
  const {
    label,
    paddingTop,
    paddingBottom,
    half,
    style,
    letterSpacing
  } = props;

  const { colors } = props.theme;

  return (
    <Text
      style={[
        {
          fontFamily: customFonts.merriweatherSansBlack,
          fontSize: getFontSize(SIZES.FONT - 1),
          fontWeight: Platform.OS === 'ios' ? 'bold' : '800',
          paddingTop: paddingTop ? paddingTop : 0,
          paddingBottom: paddingBottom ? paddingBottom : 0,
          letterSpacing: letterSpacing ? letterSpacing : 1,
          width: half ? HalfCardImageWidth - 32 : '100%',
          color: colors.sectionArticle
        },
        { ...style }
      ]}>
      {label && label.toUpperCase()}
    </Text>
  );
}

export default withTheme(SectionLabel);
