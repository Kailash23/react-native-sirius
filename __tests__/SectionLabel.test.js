import SectionLabel from '@atoms/SectionLabel';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Text } from 'react-native';

configure({ adapter: new Adapter() });

describe('Testing <SectionLabel/>', () => {
  test('SectionLabel should render properly with given title', () => {
    const wrapper = shallow(<SectionLabel label={'Test Label'} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('SectionLabel should NOT render if the label is null', () => {
    const wrapper = shallow(<SectionLabel label={null} />);
    expect(wrapper).toEqual({});
  });
});
