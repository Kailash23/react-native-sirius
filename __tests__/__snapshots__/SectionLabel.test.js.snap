// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SectionLabel/> SectionLabel should render properly with given title 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  label="Test Label"
/>
`;
